tests/has_many_test.js

```diff
@@ -0,0 +1,1633 @@
+            {
+              title: 'a', subtasks: [
+                { title: 'c' },
+                { title: 'a' },
+                { title: 'b' }
+              ]
+            }
+          ]
+        }, {
+          include: [{ association: User.Tasks, include: [Task.SubTasks] }]
+        }), User.create({
+          id: 2,
+          tasks: [
+            {
+              title: 'a', subtasks: [
+                { title: 'b' },
+                { title: 'a' },
+                { title: 'c' }
+              ]
+            },
+            {
+              title: 'c', subtasks: [
+                { title: 'a' }
+              ]
+            },
+            {
+              title: 'b', subtasks: [
+                { title: 'a' },
+                { title: 'b' }
+              ]
+            }
+          ]
+        }, {
+          include: [{ association: User.Tasks, include: [Task.SubTasks] }]
+        })]);
+
+        const users = await User.findAll({
+          include: [{
+            association: User.Tasks,
+            limit: 2,
+            order: [['title', 'ASC']],
+            separate: true,
+            as: 'tasks',
+            include: [
+              {
+                association: Task.SubTasks,
+                order: [['title', 'DESC']],
+                separate: true,
+                as: 'subtasks'
+              }
+            ]
+          }],
+          order: [
+            ['id', 'ASC']
+          ]
+        });
+
+        expect(users[0].tasks.length).to.equal(2);
+
+        expect(users[0].tasks[0].title).to.equal('a');
+        expect(users[0].tasks[0].subtasks.length).to.equal(3);
+        expect(users[0].tasks[0].subtasks[0].title).to.equal('c');
+        expect(users[0].tasks[0].subtasks[1].title).to.equal('b');
+        expect(users[0].tasks[0].subtasks[2].title).to.equal('a');
+
+        expect(users[0].tasks[1].title).to.equal('b');
+        expect(users[0].tasks[1].subtasks.length).to.equal(2);
+        expect(users[0].tasks[1].subtasks[0].title).to.equal('c');
+        expect(users[0].tasks[1].subtasks[1].title).to.equal('a');
+
+        expect(users[1].tasks.length).to.equal(2);
+        expect(users[1].tasks[0].title).to.equal('a');
+        expect(users[1].tasks[0].subtasks.length).to.equal(3);
+        expect(users[1].tasks[0].subtasks[0].title).to.equal('c');
+        expect(users[1].tasks[0].subtasks[1].title).to.equal('b');
+        expect(users[1].tasks[0].subtasks[2].title).to.equal('a');
+
+        expect(users[1].tasks[1].title).to.equal('b');
+        expect(users[1].tasks[1].subtasks.length).to.equal(2);
+        expect(users[1].tasks[1].subtasks[0].title).to.equal('b');
+        expect(users[1].tasks[1].subtasks[1].title).to.equal('a');
+      });
+
+      // Query result are non-deterministic
+      // Reason: https://github.com/sequelize/sequelize/issues/13088
+      it.skip('should fetch associations for multiple instances with limit and order and a belongsTo relation', async function() {
```

Comment:
Could we unskip this by sorting the results array afterwards?

Reply:
Sorting it afterwards just for the test to pass is not a good solution, in my opinion. I'll investigate further to see where exactly it is failing.